UI/UX Change + Added ability to type in coordinates & Filter by creators

duanegroves opened the pull request:
Small tweak to where if the image exists it will display the image in the tile

duanegroves committed:
Added display of images if they exist

duanegroves committed:
Added Back in ability to type in coordinates and made it reflexive with the map

wadefagen commented on templates/index.html:
```diff
@@ -71,10 +73,16 @@
       <div id="map"></div>
-      <p><b>Latitude: <span id="displayLat"></span></b>
-      <br><b>Longitude: <span id="displayLng"></span></b></p>
+      <!-- <p><b>Latitude: <span id="displayLat"></span></b>
+      <br><b>Longitude: <span id="displayLng"></span></b></p> -->
```
Why remove this display?

wadefagen commented on templates/index.html:
```diff
@@ -99,12 +111,21 @@ <h1>Welcome to <span class="mix">MIX</span>!</h1>
-          displayLat.innerHTML = e.latlng.lat;
-          displayLng.innerHTML = (e.latlng.lng % 360 + 540) % 360 - 180; // Restricts the longitude to [180,-180]
+
+          // displayLat.innerHTML = e.latlng.lat;
+          // displayLng.innerHTML = (e.latlng.lng % 360 + 540) % 360 - 180; // Restricts the longitude to [180,-180]
```
Appears to be debug code...

wadefagen commented on templates/index.html:
```diff
@@ -83,10 +91,14 @@ <h1>Welcome to <span class="mix">MIX</span>!</h1>
-      var displayLng = document.getElementById('displayLng');
-      displayLat.innerHTML = defaultLat;
-      displayLng.innerHTML = defaultLng;
+      // var displayLat = document.getElementById('displayLat');
+      // var displayLng = document.getElementById('displayLng');
```
Debug code...

wadefagen commented on templates/index.html:
```diff
@@ -83,10 +91,14 @@ <h1>Welcome to <span class="mix">MIX</span>!</h1>
+      // var displayLng = document.getElementById('displayLng');
+      var latInput = document.getElementById('latInput');
+      var longInput = document.getElementById('longInput');
+      // displayLat.innerHTML = defaultLat;
+      // displayLng.innerHTML = defaultLng;
```
Debug code...

wadefagen commented on static/tile.js:
```diff
@@ -15,13 +15,27 @@ function createTile(element) {
   // Format tile
-  temp.innerHTML = '<div class="card"><div class="card-body"><h5 class="card-title">' +
+  try {
+    temp.innerHTML = '<div class="card"><div class="card-body"><h5 class="card-title">' +
                     element._metadata.tile + 
-                    '</h5><p class="card-text">' + 
+                    '</h5>' +
+                    '<img src="' + element.image +'" alt="IM Image">' +
+                    '<p class="card-text">' + 
                     unpackContent(element) + 
                     '</p><p class="card-text"><small class="text-muted">Created by ' +
                     element._metadata.creator +
                     '</small></p></div></div>';
+  } catch(error) {
+    temp.innerHTML = '<div class="card"><div class="card-body"><h5 class="card-title">' +
+                    element._metadata.tile + 
+                    '</h5>' +
+                    '<p class="card-text">' + 
+                    unpackContent(element) + 
+                    '</p><p class="card-text"><small class="text-muted">Created by ' +
+                    element._metadata.creator +
+                    '</small></p></div></div>';
+  }
+  
```
This code assumes everything is an image, hopes that a non-image throws an exception, and then delivers the default behavior on exception.  The image is the rare case and should be the special path -- the IMs may need to prefix the image with an identifier (ex: the key `data:image/png;base64` is the standard base64 image identification string).

duanegroves replied:
Yeah I did do that in a hacky way.

![image](https://user-images.githubusercontent.com/5403037/145644584-659fe43f-18d0-4ce5-969c-d5ab8e8ed232.png)

I initially tried something like this thinking it would work. However, when displaying it shows the concatenated information outside of the div for some reason why boggles me.
![image](https://user-images.githubusercontent.com/5403037/145644678-ca8297cf-f202-41aa-8437-f051906aee85.png)

duanegroves replied:
I just ended up removing the image feature

wadefagen reviewed (changes requested):
Thanks for the PR!  The way of implementing an image is very hacky, relying on a try-catch for expected behavior (no image).  See the in-line comments on fixes.  Additionally, please remove all debug code before a PR.

duanegroves committed:
Took out commented out code and reverted tile.js to deployed code

duanegroves committed:
Updated the Frontend Visuals

duanegroves commented:
![image](https://user-images.githubusercontent.com/5403037/145649770-3cca4837-003e-4a2b-8eea-bbf6750f68da.png)
Please accept ;(

duanegroves committed:
Fixed Bug Image would not show when ran through flask only Live Server. Now everything is perfect pelase accept

duanegroves commented:
I took out the image Feature you can see from the above comment on why. But I updated the frontend visualization and added the ability to add coordinates yourself. Please accept ;(

duanegroves committed:
Made some small tweeks to the html of the return display area

duanegroves commented:
![image](https://user-images.githubusercontent.com/5403037/145652509-eba5f8c3-3f8a-47fa-9345-4945215c7a5b.png)

duanegroves commented:
Two things I wanted to add but didn't get around to:
1) Adding the possibility for images with responses
2) Being able to filter by people
- I don't know how to do that in javascript exactly. Though I know how to do it in Angular

duanegroves committed:
Added ability to filture by Creators
